src/cdomains/floatOps/stubs.c

```diff
@@ -0,0 +1,87 @@
+#include <stdio.h>
+#include <math.h>
+#include <float.h>
+#include <fenv.h>
+#include <assert.h>
+#include <caml/mlvalues.h>
+#include <caml/alloc.h>
```

Comment:
>It would be a lot safer to use MPFR to do all this, especially since it supports all the operations (arithmetic and standard library) with rounding mode argument on all of them: https://antoinemine.github.io/Apron/doc/api/ocaml/Mpfr.html. Also would allow analyzing long doubles.

The critical point seems to be that the computations in the abstract need not be performed on more precise float representations. From the [paper, page 6](https://www-apr.lip6.fr/~mine/publi/article-mine-esop04.pdf):
>This semantics frees the analyzer from the job of statically determining the rounding mode of the expressions and allows the analyzer to use, in the abstract, less precise formats that those used in the concrete (however, using a more precise format in the abstract remains unsound).

The abstract operations should be performed with the same precision as in the concrete. Is there a way to set the precision in MPFR? One would think so, but I didn't find anything regarding this.

Reply:
> These things shouldn't change often, but with OCaml 5.0 around the corner with a major overhaul of the language runtime, it might cause problems for us going forward.

The stubs we use here seem so basic, it is unlikely they'll break with OCaml 5.0. I'd be far more worried about advanced libraries such as apron w.r.t. to the bindings being broken.

Reply:
I don't think adding `-no-float-const-prop` would be too much harm to us, we don't heavily rely on floats anywhere but here!

Reply:
Just to have it documented here: last week at GobCon we decided to let the stubs remain for now and not block this PR. We'll have to figure out future directions ourselves later.